plugins/localization-connector/src/components/contexts/builderContext.tsx

```diff
@@ -0,0 +1,183 @@
+    <BuilderContext.Provider value={props.context}>
+      {props.children}
+    </BuilderContext.Provider>
+  )
+}
+
+export const getContext = (): any => {
+  const builderContext = useContext(BuilderContext)
+  if (builderContext === undefined) {
+    const caller = getContext.caller.name
+    throw new Error(`${caller} must be used within a BuilderProvider`)
+  }
+  return builderContext
+}
+
+export const isLocaleNotEligibleToLocaliseFrom = (): boolean => {
+  const disallow = true
+  const allow = false
+
+  const allowedLocalesToTranslateFrom = getTranslatableSourceLocales()
+  if (allowedLocalesToTranslateFrom === undefined) return allow
+
+  const sourceLocale = getSourceLocale()
+  if (sourceLocale === undefined) return disallow
+
+  const isIncluded = allowedLocalesToTranslateFrom.includes(sourceLocale)
+
+  if (isIncluded) return allow
+  return disallow
+}
+
+export const modelHasOnlyOneLocale = (): boolean => {
+  const hasNoTargetLocales = true
+
+  const targetLocales = getTargetLocales()
+
+  if (targetLocales == undefined) return hasNoTargetLocales
+  return targetLocales.length === 0
+}
+
+export const pageHasNoBlocks = (): boolean => {
+  const withoutBlocks = true
+  const pageBlocks = getPageBlocks()
+
+  if (pageBlocks === undefined) return withoutBlocks
+
+  return pageBlocks.length === 0
+}
+
+const getPageBlocks = (): any | undefined => {
+  try {
+    const ctx = getContext()
+    const blocks = ctx.designerState.editingContentModel.data
+      .get('blocks')
+      .toJSON()
+    return blocks
+  } catch {
+    return undefined
+  }
+}
+
+const getTranslatableSourceLocales = (): Array<string> | undefined => {
+  try {
+    const ctx = getContext()
+    return ctx.designerState.editingContentModel.model.fields
+      .find((field: any) => field.name === 'allowedLocales')
+      .enum.toJSON()
+  } catch {
+    return undefined
+  }
+}
+
+export const getSourceLocale = (): string | undefined => {
+  try {
+    const ctx = getContext()
+    return ctx.designerState.editingContentModel.data.toJSON().locale
+  } catch {
+    return undefined
+  }
+}
+
+export const getTargetLocales = (): Array<string> | undefined => {
+  try {
+    const ctx = getContext()
+    const sourceLocale = getSourceLocale()
+    const targetLocales = ctx.designerState.editingContentModel.model.fields
+      .find((field: any) => field.name === 'locale')
+      .enum.toJSON()
+
+    if (targetLocales.includes(sourceLocale)) {
+      targetLocales.splice(targetLocales.indexOf(sourceLocale), 1)
+    }
+    return targetLocales
+  } catch {
+    return undefined
+  }
+}
+
+type MemsourceArgs = {
+  memsourceToken: string
+  sourceLocale: string
+  projectName: string
+  payload: any
+  memsourceInputSetting: string | undefined
+}
+
+export const getMemsourceArguments = (): MemsourceArgs | undefined => {
+  const memsourceToken = getMemsourceToken()
+  if (memsourceToken === undefined) {
+    alert('Cannot find memsourceToken')
+    return undefined
```

Comment:
> I'm probably doing something wrong but I'm getting CORS errors from `memsource` when using the plugin

Yeah, that's an issue with their API, their response and I quote: 

> It seems you are calling that from a browser (website).
> 
> It is unfortunately not possible and results in that error.
> 
> If you would use a script to send that call, it would work.

I've already answered them to see if we can move forward. Until then, I would say this pull request is on hold...

Reply:
I really like this!

Reply:
Hope you like the new adaptation :)